internal/clients/client.go

```diff
@@ -122,18 +127,35 @@ func NewClient(config ClientConfig) (*Client, error) {
 		opts = append(opts, hcpConfig.WithClientCredentials(config.ClientID, config.ClientSecret))
 	} else if config.CredentialFile != "" {
 		opts = append(opts, hcpConfig.WithCredentialFilePath(config.CredentialFile))
-	} else if config.WorloadIdentityTokenFile != "" && config.WorkloadIdentityResourceName != "" {
+	} else if (config.WorkloadIdentityToken != "" || config.WorkloadIdentityTokenFile != "") && config.WorkloadIdentityResourceName != "" {
```

Comment:
This is hard to parse for my out-of-context brain. Is it possible to do it better?

Reply:
I've pulled out the combined or logic into a standalone variable, and added some additional comments explaining the flow. Hopefully, that is better?